nautobot/dcim/models/devices.py

```diff
@@ -408,12 +430,65 @@ class Platform(OrganizationalModel):
     )
     description = models.CharField(max_length=200, blank=True)
 
+    @cached_property
+    def _network_drivers(self):
+        NETWORK_DRIVERS = deepcopy(get_settings_or_config("NETWORK_DRIVERS"))
+        for key, mapping in (
+            ("ansible", ANSIBLE_LIB_MAPPER_REVERSE),
+            # TODO ("hier_config", HIER_CONFIG_LIB_MAPPER_REVERSE),
+            ("netmiko", NETMIKO_LIB_MAPPER_REVERSE),
+            ("ntc_templates", NTCTEMPLATES_LIB_MAPPER_REVERSE),
+            ("pyats", PYATS_LIB_MAPPER_REVERSE),
+            ("pyntc", PYNTC_LIB_MAPPER_REVERSE),
+            ("scrapli", SCRAPLI_LIB_MAPPER_REVERSE),
+        ):
+            # Fill in values from *LIB_MAPPER_REVERSE but let NETWORK_DRIVERS override them
+            NETWORK_DRIVERS[key] = {**mapping, **(NETWORK_DRIVERS.get(key, {}))}
+        return NETWORK_DRIVERS
+
+    @property
+    def ansible_driver(self):
+        """Collection string for use with Ansible; derived from network_driver by netutils."""
+        return self._network_drivers["ansible"].get(self.network_driver, None)
+
+    # TODO, need netutils 1.5.0 release first
+    # @property
+    # def hier_config_driver(self):
+    #     """Style string for use with hier_config library; derived from network_driver by netutils."""
+    #     return self._network_drivers["hier_config"].get(self.network_driver, None)
+
+    @property
+    def netmiko_driver(self):
+        """Driver for use with netmiko library; derived from network_driver by netutils."""
+        return self._network_drivers["netmiko"].get(self.network_driver, None)
+
+    @property
+    def ntc_templates_driver(self):
+        """Platform string for use with ntc-templates parsers; derived from network_driver by netutils."""
+        return self._network_drivers["ntc_templates"].get(self.network_driver, None)
+
+    @property
+    def pyats_driver(self):
+        """OS string for use with PyATS library; derived from network_driver by netutils."""
+        return self._network_drivers["pyats"].get(self.network_driver, None)
+
+    @property
+    def pyntc_driver(self):
+        """Device type string for use with pyntc library; derived from network_driver by netutils."""
+        return self._network_drivers["pyntc"].get(self.network_driver, None)
+
+    @property
+    def scrapli_driver(self):
+        """Platform string for use with scrapli library; derived from network_driver by netutils."""
+        return self._network_drivers["scrapli"].get(self.network_driver, None)
+
```

Comment:
Is the plan to remove the concrete `napalm_driver` field and convert to use this pattern in 2.0?

Reply:
The linked comment appears to be suggesting the API remains on the platform model. My understanding is that from the device object's perspective you would access either the `self.platform.network_driver` concrete field which would return a string or you would access the `self.platform.library_network_drivers` property which would return a dict. Is that correct?

```py
>>> device_object = Device.objects.first()
>>> device_object.platform.network_driver
'arista_eos'
>>> device_object.platform.library_network_drivers
{
    'ansible': 'arista.eos.eos',
    'hier_config': 'eos',
    'netmiko': 'arista_eos',
    'ntc_templates': 'arista_eos',
    # pyats omitted because no match was found or pyats: None?
    'pyntc': 'arista_eos_eapi',
    'scrapli': 'arista_eos',
}
```

Reply:
That is my suggestion. @lampwins @itdependsnetworks care we weigh in?

I'd like to see `network_driver` have a strong hint that it's the netutils normalized driver name.